### DISCUSSION TITLE
validation number

### DISCUSSION THREAD

**Discussion by samaremad** [serenity-is/Serenity#5356] [Help] [answered]
hello everyone

i try to make a validation for a field 
i want the form dont accept the number in that field if this number not 10 digit

i write the following code in the dialog.ts but it dosent work correctly 
can you please help me with that:

this.form.Username.addValidationRule(this.uniqueName, x => {
                var nationalID = this.form.Username.element.length;
                                             
                if (nationalID !=10) {
                    return "National ID is too short"
                }
            })

**Comment by edwardch** [comment]
Check out the password length validation in [ChangePasswordPanel.ts](https://github.com/serenity-is/Serene/blob/8f0f778d1331d3988610854e0a85e14dbff20757/Serene/Serene.Core/Modules/Membership/Account/ChangePassword/ChangePasswordPanel.ts#L15).

You say it is a number though, so maybe try add the `IntegerEditor` option to this field in your xyzRow.cs like so:
```cs
[IntegerEditor(MinValue = 1000000000)]
[DisplayName("National ID"), Size(10), NotNull, QuickSearch]
public Int64? NationalID
{
    get { return Fields.NationalID[this]; }
    set { Fields.NationalID[this] = value; }
}
```

**Reply by samaremad** [reply]
dosnt work also

**Comment by edwardch** [comment, marked as answer]
And if you change your validation rule to check the length of the **value**, and not the element (like in the password length validator)?

```ts
this.form.Username.addValidationRule(this.uniqueName, x => {
// Your example is using this.form.Username.element.length
var nationalID = this.form.Username.value.length;

            if (nationalID !=10) {
                return "National ID is too short"
            }
        })
```

**Reply by samaremad** [reply]
this.form.Username.change( x=> {
                var nationalID = this.form.Username.value.toString.length;
                console.log(nationalID);
            })
            
  i write this because only when it is string it take value.lenght
  and in console the length appear 1 whatever i write

**Reply by edwardch** [reply]
toString is a function and needs the brackets to be evaluated... 

Check these outputs:
```ts
this.form.Username.change(x => {
    console.log("Not working");
    console.log("this.form.Username.value.toString.length");
    console.log(this.form.Username.value.toString.length);
    console.log("Working");
    console.log("this.form.Username.value.toString().length");
    console.log(this.form.Username.value.toString().length);
    console.log("this.form.Username.value.length");
    console.log(this.form.Username.value.length);
})
```

**Reply by samaremad** [reply]
thank you soooooooooooo much it work correctly

**Reply by edwardch** [reply]
Great! Please mark as answer to help anyone else facing a similar issue. Thanks

**Reply by samaremad** [reply]
thank you again and if you have knowledge in (entity data model ) can you please help me with another issue :D

**Reply by edwardch** [reply]
Sure, add a discussion and see what we can do
